crates/transport/src/layers/fallback.rs

```diff
@@ -0,0 +1,447 @@
+use std::{
+    collections::{BinaryHeap, VecDeque},
+    num::NonZeroUsize,
+    ops::{Deref, DerefMut},
+    sync::{Arc, Mutex},
+    task::{Context, Poll},
+    time::{Duration, Instant},
+};
+
+use alloy_json_rpc::{RequestPacket, ResponsePacket};
+use futures::{stream::FuturesUnordered, StreamExt};
+use tower::{Layer, Service};
+use tracing::{debug, trace};
+
+use crate::{TransportError, TransportErrorKind, TransportFut};
+
+// Constants for the ranking algorithm
+const STABILITY_WEIGHT: f64 = 0.7;
+const LATENCY_WEIGHT: f64 = 0.3;
+const DEFAULT_SAMPLE_COUNT: usize = 10;
+const DEFAULT_ACTIVE_TRANSPORT_COUNT: usize = 3;
+
+/// Represents performance metrics for a transport.
+#[derive(Debug)]
+struct TransportMetrics {
```

Comment:
can we move this below the main type

makes it easier to navigate if the main type is top of file